File: api/contest.go
package api

import (
	"main.go/controller"

	"github.com/gin-gonic/gin"
)

func ContestRoutes(router *gin.RouterGroup) {
	contestController := controller.NewContestController()
	router.POST("/contests", contestController.CreateContest)
	router.GET("/contests/:id", contestController.GetContest)
} 

File: routes/contest.ts
import express from 'express';
import ContestController from '../controller/Contest';
import { asyncHandler } from '../utils/asyncHandler';
import { validateToken } from '../middleware/auth';

if(process.env.NODE_ENV !== 'production') {
    require('dotenv').config();
}

const router = express.Router();
const contestController = new ContestController();

router.get('', asyncHandler(contestController.getContest.bind(contestController)));

router.post('/login', asyncHandler(contestController.login.bind(contestController)));
router.post('/submit', validateToken as express.RequestHandler, asyncHandler(contestController.submit.bind(contestController)));
router.post('/create', asyncHandler(contestController.createContest.bind(contestController)));

router.get('/',
    asyncHandler(
        async (req, res) => {
            return res.status(200).json({ message: 'Contest API' });
        }
    )
);

export default router;

